### ISSUE TITLE
New rule: `await-user-event`

### ISSUE THREAD

**Issue by bennettdams** [testing-library/eslint-plugin-testing-library#647]
### Name for new rule

await-user-event

### Description of the new rule

Now that the `userEvent`'s methods are all async (with `v14`), wouldn't it make sense to add a rule to enfore `await` when using them? 

A similar rule [already exists](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/await-fire-event.md) for awaiting `fireEvent`.

### Testing Library feature

All usage of `userEvent` with `v14`.

### Testing Library framework(s)

I guess all of them?

### What category of rule is this?

Warns about a potential error

### Code examples

```ts
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
...
// click the button to reload the page
const reloadPageButton = screen.getByText('Reload page')
userEvent.click(reloadPageButton)
```

Instead, it should enforce using `await`:

```diff
- userEvent.click(reloadPageButton)
+ await userEvent.click(reloadPageButton)
```

### Anything else?

I'm not sure if this is even possible to integrate, as this rule should only be active for `userEvent` usage of `v14`.

### Do you want to submit a pull request to make the new rule?

No
